packages/command-palette/src/use-command-palette.tsx

```diff
@@ -188,6 +188,27 @@ export const useCommandPalette = ( {
 		return filter?.( site ) && hasCapability;
 	};
 
+	const trackSelectedCommand = ( command: Command ) => {
+		recordTracksEvent( 'calypso_hosting_command_palette_command_select', {
+			command: command.name,
+			has_nested_commands: !! command.siteFunctions,
+			list_count: commands.length,
+			list_visible_count: listVisibleCount,
+			current_route: currentRoute,
+			search_text: search,
+		} );
+	};
+
+	const trackCompletedCommand = ( command: Command ) => {
+		recordTracksEvent( 'calypso_hosting_command_palette_command_complete', {
+			command: command.name,
+			list_count: commands.length,
+			list_visible_count: listVisibleCount,
+			current_route: currentRoute,
+			search_text: search,
+		} );
+	};
+
```

Comment:
[Nitpick]
We could improve this a bit by composing a higher-order function and having a base.
Something like this:
```
const trackCommand = (
		command: Command,
		eventName = 'calypso_hosting_command_palette_command_complete',
		extra = {}
	) => {
		recordTracksEvent( eventName, {
			command: command.name,
			list_count: commands.length,
			list_visible_count: listVisibleCount,
			current_route: currentRoute,
			search_text: search,
			...extra,
		} );
	};

	const trackSelectedCommand = ( command: Command ) => {
		trackCommand( command, 'calypso_hosting_command_palette_command_select', {
			has_nested_commands: !! command.siteFunctions,
		} );
	};
```
Or any other variation, like not creating `trackSelectedCommand` and explicitly calling the main function.